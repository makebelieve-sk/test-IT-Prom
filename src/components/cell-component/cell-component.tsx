import React, { useState } from 'react';

type CellComponentType = {
    handleSubmit: (e, el: string, i: number, j: number) => any, 
    editMode: string, 
    elem: any, 
    j: number, 
    i: number, 
    clickProp: any
};

export const CellComponent: React.FC<CellComponentType> = ({ handleSubmit, editMode, elem, j, i, clickProp }) => {
    const [value, setValue] = useState<string>(elem);

    return (
        <>
            <td key={`${elem}--${j}`} onClick={clickProp ? clickProp : null} className="table-td">
                { editMode === 'hidden' ? elem : null }

                <form 
                    onSubmit={(e) => {
                        elem = value;
                        handleSubmit(e, elem, i, j);
                    }}
                    className={`edit-mode-${editMode}`}  
                    key={`${elem}-${j}-form`}
                >
                    <input 
                        type="text"
                        value={value}
                        onChange={(e) => {
                            setValue(e.target.value);
                        }}
                    />
                </form>
            </td>
        </>
    )
}