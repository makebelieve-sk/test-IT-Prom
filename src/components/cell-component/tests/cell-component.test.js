import React from 'react';
import ReactDOM from 'react-dom'; 
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { CellComponent } from '../cell-component';

Enzyme.configure({ adapter: new Adapter() });

const mock = {
    elem: `some string`,
    editMode: `editMode`,
    clickProp: jest.fn(),
    j: 2,
    i: 1,
    handleSubmit: jest.fn()
};

it(`CellComponent correctly rendering`, () => {
    const tree = renderer  
        .create(<CellComponent 
            elem = {mock.elem}
            editMode = {mock.editMode}
            clickProp = {mock.clickProp}
            handleSubmit = {mock.handleSubmit}
            i = {mock.i}
            j = {mock.j}
        />)
        .toJSON();

    expect(tree).toMatchSnapshot();
});

describe('CellComponent correctly rendering and functioning', () => {
    let wrapper;
  
    beforeEach(() => {
        wrapper = shallow(<CellComponent 
            elem = {mock.elem}
            editMode = {mock.editMode}
            clickProp = {mock.clickProp}
            handleSubmit = {mock.handleSubmit}
            i = {mock.i}
            j = {mock.j}
        />);
    });

    it(`+++ CellComponent renders without crashing`, () => {
        const div = document.createElement('div');
      
        ReactDOM.render(<CellComponent />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('+++ render the connected CellComponent', () => {
        expect(wrapper.length).toEqual(1);
    });
});