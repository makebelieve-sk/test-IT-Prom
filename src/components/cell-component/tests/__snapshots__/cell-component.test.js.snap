// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CellComponent correctly rendering 1`] = `
<td
  onClick={[MockFunction]}
>
  <form
    className="edit-mode-editMode"
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
      value="some string"
    />
  </form>
</td>
`;
